libs/shader/sim/shader.ts

```diff
@@ -0,0 +1,37 @@
+/// <reference path="../../screen/sim/image.ts" />
+
+namespace pxsim.ShaderMethods {
+    export function _mergeImage(dst: RefImage, src: RefImage, xy: number) {
+        mergeImage(dst, src, pxsim.ImageMethods.XX(xy), pxsim.ImageMethods.YY(xy));
+    }
+
+    function mergeImage(dst: RefImage, src: RefImage, x0: number, y0: number) {
```

Comment:
Just wondering, why reimplement this? Could you call `pxsim.ImageMethods.mergeImage/mapImage` here instead?

Reply:
this is the simulator implementation!

Reply:
these apis are the ones i'm adding, so they don't already exist in imagemethods

Reply:
Would this be equivalent?
```ts
    function mergeImage(dst: RefImage, src: RefImage, x0: number, y0: number) {
        pxsim.ImageMethods.mergeImage(dst, src, x0, y0);
    }
```